File: apps/java/io/dustin/apps/common/model/ResponseWithScroll.java
package io.dustin.apps.common.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Getter
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class ResponseWithScroll<T> {
    private T result;
    private boolean isLast;
    private Long nextId;

    public static <T> ResponseWithScroll<T> from(T result, boolean isLast, Long nextId) {
        return new ResponseWithScroll<>(result, isLast, nextId);
    }
}

File: apps/src/main/kotlin/io/dustin/apps/common/model/ResponseWithScroll.kt
package io.dustin.apps.common.model

data class ResponseWithScroll<T>(
    val result: T,
    val isLast: Boolean,
    val nextId: Long?
) {
    companion object {
        fun <T> from(result: T, isLast: Boolean, nextId: Long?): ResponseWithScroll<T> {
            return ResponseWithScroll(result, isLast, nextId)
        }
    }
}
